commands/generate.go

```diff
@@ -1,23 +1,118 @@
 package commands
 
 import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
 	"github.com/spf13/cobra"
 )
 
 func GenerateCmd() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "generate",
+		Use:   "generate <file name>",
 		Short: "Creates new migrations",
+		Long:  "morph generate create_users --driver postgresql --dir db/migrations --timestamp",
 		Args:  cobra.ExactArgs(1),
 		Run:   generateCmdF,
 	}
 
-	cmd.Flags().StringP("driver", "d", "", "the driver to use")
-	cmd.MarkFlagRequired("driver")
+	cmd.Flags().StringP("driver", "d", "", "the driver to use.")
+	cmd.Flags().BoolP("timestamp", "t", false, "a timestamp prefix will be added to migration file if set.")
+	cmd.Flags().StringP("timeformat", "f", "unix", "timestamp format to be used for timestamps.")
+	cmd.Flags().StringP("timezone", "z", "utc", "time zone to be used for timestamps.")
+	cmd.Flags().BoolP("sequence", "s", false, "a sequence number prefix will be added to migration file if set.")
+	cmd.Flags().IntP("number-of-digits", "n", 6, "number of digits to be used in sequence prefix.")
+	_ = cmd.MarkFlagRequired("driver")
 
 	return cmd
 }
 
-func generateCmdF(cmd *cobra.Command, _ []string) {
-	cmd.Println("To Be Implemented")
+func generateCmdF(cmd *cobra.Command, args []string) {
+	dir, _ := cmd.Flags().GetString("dir")
+	extention := "sql"
```

Comment:
Since we know the driver, i guess we can infer the extension from it right? So instead of hardcoding it we can create a lookup table that will give us the extension from the driver name. Thoughts?

Reply:
Makes sense 👍